src/views/SingleMovie.vue

```diff
@@ -0,0 +1,148 @@
+<script setup lang="ts">
+import { computed } from "vue";
+import BreadCrumb from "@/components/BreadCrumb.vue";
+import { useMovieStore } from "@/stores/movies";
+import { useRoute } from "vue-router";
+import lengthToTime from "@/helpers/lengthToTime";
+const movieStore = useMovieStore();
+const route = useRoute();
+
+const selectedId = computed(() => route.params.id.toString());
+const selectedMovie = computed(
+  () =>
+    movieStore.movieList.find(
+      (movie) => movie.id.toString() === selectedId.value
+    ) || {
+      title: "Movie.title",
+      length: 115,
+      description: "description...",
+      poster_url:
+        "https://upload.wikimedia.org/wikipedia/en/8/8e/Dune_%282021_film%29.jpg",
+      genre: { name: "genre" },
+    }
```

Comment:
Is this data placeholder something temporary? 🤔

Reply:
It was initially from testing and making the component. I left it behind as it was a lazy way of preventing errors while data is not loaded. I should really go back and define those as optional.